src/ec2/model.ts

```diff
@@ -93,6 +95,19 @@ export class Ec2ConnectionManager {
         throw new ToolkitError(generalErrorMessage + message, errorInfo)
     }
 
+    private async pollForPermissions(roleArn: string): Promise<void> {
+        const timeout = new Timeout(-1)
+        await showMessageWithCancel(`Adding inline policy to IAM role ${roleArn}`, timeout)
+        while (true) {
+            const permissionsAdded = await this.hasProperPermissions(roleArn)
+            if (permissionsAdded) {
+                break
+            }
+            await sleep(1000)
+        }
```

Comment:
what if `hasProperPermissions` never succeeds? the while-loop needs to break if the `timeout` was canceled.


    while(!timeout.completed)

Reply:
Currently not using this method since it appeared to give the same problems. That is, the API would say the permissions are attached, but terminal would crash. I still updated it in case there is a way to modify it to avoid this.